src/kschmaus1.java

```diff
@@ -0,0 +1,175 @@
+					postFix.append(op.pop()).append(' ');					//checks precedence against mapping
+				}
+				op.push(token);
+			}
+			else if (token.equals("(")) {		//pushes opening bracket to stack
+				op.push(token);
+			}
+			else if (token.equals(")")) {
+				while (!op.peek().equals("(")) {			//if closing bracket is found pop operators to StringBuilder until 
+					postFix.append(op.pop()).append(' ');	//opening bracket is found, then discard.
+				}
+				op.pop();
+			}
+			else {
+				postFix.append(token).append(' ');
+			}
+		}
+		while(!op.isEmpty()) {
+			postFix.append(op.pop()).append(' ');			//pop rest of operators to StringBuilder 
+		}
+		return postFix.toString();
+	}
+	
+	public String calcPostFixLong(String postFix) {		//calculate result if it is a long
+		Stack<String> numbers = new Stack<>();
+		String[] splitPost = postFix.split(" ");
+		Calculations calc = new Calculations();
+		long result = 0;
+		String temp = "";
+		
+		for (int i = 0; i < splitPost.length; i++) {
+			String currItem = splitPost[i];
+			
+			if (!operatorMap.containsKey(currItem)) {
+				numbers.push(currItem);
+			}
+			else {
+				temp = numbers.pop();
+				long val1 = Long.parseLong(temp);
+				temp = numbers.pop();
+				long val2 = Long.parseLong(temp);
+				
+				switch(currItem) {
+					case "+":
+						result = calc.add(val2, val1);
+						numbers.push(Long.toString(result));
+						break;
+						
+					case "-":
+						result = calc.returnDiff(val2, val1);
+						numbers.push(Long.toString(result));
+						break;
+					
+					case "/":
+						result = calc.divide(val2, val1);
+						numbers.push(Long.toString(result));
+						break;
+			
+					case "*":
+						result = calc.multiply(val2, val1);
+						numbers.push(Long.toString(result));
+						break;
+				}
+			}
+		}
+		return numbers.pop();
+	}
+	
+	public String calcPostFixDouble(String postFix) {
+		Stack<String> numbers = new Stack<>();
+		String[] splitPost = postFix.split(" ");
+		Calculations calc = new Calculations();
+		double result = 0.0;
+		String temp = "";
+		
+		for (int i = 0; i < splitPost.length; i++) {
+			String currItem = splitPost[i];
+			
+			if (!operatorMap.containsKey(currItem)) {
+				numbers.push(currItem);
+			}
+			else {
+				temp = numbers.pop();
+				double val1 = Double.parseDouble(temp);
+				temp = numbers.pop();
+				double val2 = Double.parseDouble(temp);
+				
+				switch(currItem) {
+					case "+":
+						result = calc.add(val2, val1);
+						numbers.push(Double.toString(result));
+						break;
+						
+					case "-":
+						result = calc.returnDiff(val2, val1);
+						numbers.push(Double.toString(result));
+						break;
+					
+					case "/":
+						result = calc.divide(val2, val1);
+						numbers.push(Double.toString(result));
+						break;
+						
+					case "*":
+						result = calc.multiply(val2, val1);
+						numbers.push(Double.toString(result));
+						break;
+				}
+			}
+		}
+		return numbers.pop();
+	}
+	
+	public static void main(String[] args) {
```

Comment:
Typically, should choose a different start point other than main. In other words, you can construct the class and then just call the class to process an input.